Bump axios from 0.27.2 to 1.1.0

- related to https://github.com/jhipster/jhipster-lite/pull/3843

The following test failed:

```
 FAIL  src/test/javascript/spec/module/secondary/RestModulesRepository.spec.ts
  ● Test suite failed to run

    src/main/webapp/app/module/secondary/RestModulesRepository.ts:59:5 - error TS2322: Type 'string | undefined' is not assignable to type 'string'.
      Type 'undefined' is not assignable to type 'string'.

    59     filename: response.headers['x-suggested-filename'],
           ~~~~~~~~

      src/main/webapp/app/module/domain/Project.ts:4:3
        4   filename: ProjectFilename;
            ~~~~~~~~
        The expected type comes from property 'filename' which is declared here on type 'Project'
```

--- a/package.json
+++ b/package.json
@@ -44,7 +44,7 @@
     "webapp:serve": "vite"
   },
   "dependencies": {
-    "axios": "0.27.2",
+    "axios": "1.1.2",
     "mitt": "3.0.0",
     "vue": "3.2.40",
     "vue-router": "4.1.5"

--- a/src/main/webapp/app/common/domain/Optional.ts
+++ b/src/main/webapp/app/common/domain/Optional.ts
@@ -19,7 +19,7 @@ export abstract class Optional<Value> {
   abstract flatMap<Output>(mapper: (feature: Value) => Optional<Output>): Optional<Output>;
   abstract orElse(value: Value): Value;
   abstract orElseGet(factory: () => Value): Value;
-  abstract orElseThrow(): Value;
+  abstract orElseThrow<U = Error>(throwable?: () => U): Value;
   abstract filter(predicate: (value: Value) => boolean): Optional<Value>;
   abstract isEmpty(): boolean;
   abstract isPresent(): boolean;
@@ -44,8 +44,11 @@ class EmptyOptional<Value> extends Optional<Value> {
     return factory();
   }
 
-  orElseThrow(): Value {
-    throw new Error("Can't get value from an empty optional");
+  orElseThrow<U>(throwable?: () => U): Value {
+    if (throwable === undefined) {
+      throw new Error("Can't get value from an empty optional");
+    }
+    throw throwable();
   }
 
   filter(): Optional<Value> {

--- a/src/main/webapp/app/module/secondary/RestModulesRepository.ts
+++ b/src/main/webapp/app/module/secondary/RestModulesRepository.ts
@@ -14,6 +14,7 @@ import { mapToLandscape, RestLandscape } from './RestLandscape';
 import { ProjectHistory } from '../domain/ProjectHistory';
 import { ModulesToApply } from '../domain/ModulesToApply';
 import { RestModulesToApply, toRestModulesToApply } from './RestModulesToApply';
+import { Optional } from '@/common/domain/Optional';
 
 export class RestModulesRepository implements ModulesRepository {
   constructor(private axiosInstance: AxiosHttp) {}
@@ -54,9 +55,9 @@ export class RestModulesRepository implements ModulesRepository {
   }
 }
 
-const mapToProject = (response: AxiosResponse<ArrayBuffer>): Project => {
-  return {
-    filename: response.headers['x-suggested-filename'],
-    content: response.data,
-  };
-};
+const mapToProject = (response: AxiosResponse<ArrayBuffer>): Project => ({
+  filename: Optional.ofUndefinable(response.headers['x-suggested-filename']).orElseThrow(
+    () => new Error('Impossible to download file without filename')
+  ),
+  content: response.data,
+});

--- a/src/test/javascript/spec/common/domain/Optional.spec.ts
+++ b/src/test/javascript/spec/common/domain/Optional.spec.ts
@@ -82,12 +82,16 @@ describe('Optional', () => {
   });
 
   describe('Or else throw', () => {
-    it('Should throw erroer for empty optional', () => {
+    it('Should throw error for empty optional', () => {
       expect(() => Optional.empty().orElseThrow()).toThrowError();
     });
 
     it('Should get value for valuated optional', () => {
       expect(Optional.of('value').orElseThrow()).toBe('value');
+    });
+
+    it('Should get supplied error when present', () => {
+      expect(() => Optional.empty().orElseThrow(() => new Error('Supplied error'))).toThrow(new Error('Supplied error'));
     });
   });
 

--- a/src/test/javascript/spec/module/secondary/RestModulesRepository.spec.ts
+++ b/src/test/javascript/spec/module/secondary/RestModulesRepository.spec.ts
@@ -106,6 +106,14 @@ describe('Rest modules repository', () => {
     expect(project.filename).toBe('file.zip');
     expect(project.content).toEqual([1, 2, 3]);
   });
+
+  it('Should fail to download when there is no suggested filename', async () => {
+    const axiosInstance = stubAxiosHttp();
+    const repository = new RestModulesRepository(axiosInstance);
+    axiosInstance.get.resolves({ headers: {}, data: [1, 2, 3] });
+
+    await expect(repository.download('path/to/project')).rejects.toEqual(new Error('Impossible to download file without filename'));
+  });
 });
 
 const restModules = (): RestModules => ({

